UnitTests/Schemas/IFC4X1/mapped-shape-with-multiple-items/src/mapped-shape-with-multiple-items.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+	Copyright (c) 2020 Technical University of Munich
+	Chair of Computational Modeling and Simulation.
+
+	TUM Open Infra Platform is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License Version 3
+	as published by the Free Software Foundation.
+
+	TUM Open Infra Platform is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <reader/IFC4X1Reader.h>
+#include <namespace.h>
+
+#include <VisualTest.h>
+
+#include <IfcGeometryConverter/ConverterBuw.h>
+#include <IfcGeometryConverter/IfcImporter.h>
+#include <IfcGeometryConverter/IfcImporterImpl.h>
+
+using namespace testing;
+
+class MappedShapeWithMultipleItems : public VisualTest {
+protected:
+
+	// Test standard values
+	buw::Image4b _background = buw::Image4b(0, 0);
+
+	virtual void SetUp() override {
+		VisualTest::SetUp();
+
+		express_model = OpenInfraPlatform::IFC4X1::IFC4X1Reader::FromFile(filename.string());
+
+		importer = buw::makeReferenceCounted<oip::IfcImporterT<emt::IFC4X1EntityTypes>>();
+		importer->collectGeometryData(express_model);
+		oip::ConverterBuwT<emt::IFC4X1EntityTypes>::createGeometryModel(model, importer->getShapeDatas());
+
+		_background = renderer->captureImage();
+		renderer->setModel(model);
+	}
+
+	virtual void TearDown() override {
+		express_model.reset();
+		VisualTest::TearDown();
+	}
+
+	virtual std::string TestName() const { return "mapped-shape-with-multiple-items"; }
+	virtual std::string Schema() const { return "IFC4X1"; }
+
+	const boost::filesystem::path filename = dataPath("mapped-shape-with-multiple-items.ifc");
+
+	std::shared_ptr<oip::EXPRESSModel> express_model = nullptr;
+	buw::ReferenceCounted<oip::IfcImporterT<emt::IFC4X1EntityTypes>> importer = nullptr;
+	buw::ReferenceCounted<oip::IfcGeometryModel> model = buw::makeReferenceCounted<oip::IfcGeometryModel>();
+};
+
+TEST_F(MappedShapeWithMultipleItems, AllEntitiesAreRead) {
+	EXPECT_THAT(express_model->entities.size(), Eq(50));
+}
+
+TEST_F(MappedShapeWithMultipleItems, IFCHasAnEssentialEntity) {
+	auto result = std::find_if(express_model->entities.begin(), express_model->entities.end(), [](auto &pair) -> bool { return pair.second->classname() == "IFCMAPPEDITEM"; });
+	EXPECT_NE(result, express_model->entities.end());
```

Comment:
Please also add tests for these entities:
- IFCEXTRUDEDAREASOLID
- IFCRECTANGLEPROFILEDEF
- IFCCARTESIANTRANSFORMATIONOPERATOR3DNONUNIFORM

Also, please use `std::count_if` in a similar way, to check that there are 4 IfcMappedItem-s as well as 4 IfcCartesianTransf...-s. This hsould be a new `TEST_F`.

Reply:
Resolved: afcbbd4